modules/fedimint-lnv2-client/src/send_sm.rs

```diff
@@ -0,0 +1,338 @@
+                        move |dbtx, preimage, old_state| {
+                            Box::pin(Self::transition_preimage(
+                                dbtx,
+                                gc_preimage.clone(),
+                                old_state,
+                                preimage,
+                            ))
+                        },
+                    ),
+                ]
+            }
+            SendSMState::Refunding(..) => {
+                vec![]
+            }
+            SendSMState::Success(..) => {
+                vec![]
+            }
+            SendSMState::Rejected(..) => {
+                vec![]
+            }
+        }
+    }
+
+    fn operation_id(&self) -> OperationId {
+        self.operation_id
+    }
+}
+
+impl SendStateMachine {
+    async fn await_funding(
+        global_context: DynGlobalClientContext,
+        txid: TransactionId,
+    ) -> Result<(), String> {
+        global_context.await_tx_accepted(txid).await
+    }
+
+    async fn transition_funding(
+        result: Result<(), String>,
+        old_state: SendStateMachine,
+    ) -> SendStateMachine {
+        let funding = match old_state.state {
+            SendSMState::Funding(funding) => funding,
+            _ => panic!("Unexpected prior state"),
+        };
+
+        SendStateMachine {
+            operation_id: old_state.operation_id,
+            contract: old_state.contract,
+            state: match result {
+                Ok(()) => SendSMState::Funded(Funded {
+                    gateway_api: funding.gateway_api,
+                    invoice: funding.invoice,
+                    refund_keypair: funding.refund_keypair,
+                }),
+                Err(error) => SendSMState::Rejected(error),
+            },
+        }
+    }
+
+    async fn gateway_send_payment(
+        gateway_api: SafeUrl,
+        federation_id: FederationId,
+        contract: OutgoingContract,
+        invoice: Bolt11Invoice,
+    ) -> Result<[u8; 32], Signature> {
+        loop {
+            match Self::try_gateway_send_payment(
+                gateway_api.clone(),
+                federation_id,
+                contract.clone(),
+                invoice.clone(),
+            )
+            .await
+            {
+                Ok(gateway_response) => match gateway_response {
+                    Ok(gateway_response) => {
+                        if contract.verify_gateway_response(&gateway_response) {
+                            return gateway_response;
+                        } else {
+                            error!("Invalid gateway response: {gateway_response:?}");
+                        }
+                    }
+                    Err(error) => {
+                        error!("Gateway returned error: {error}")
+                    }
+                },
+                Err(error) => {
+                    error!("Error while trying to reach gateway: {error}");
```

Comment:
Just a thought: we might want to retry faster if we receive a 404 error code from the gateway vs a different error code (we do this in the legacy pay in this PR https://github.com/fedimint/fedimint/pull/4024/files#diff-7e39423fc03d68d7d9b2c7f7799175ee71df89d5043d053b77f7d8067228bf42)